tree-wide: use LooseVersion instead of StrictVersion

flokli opened the pull request:
According to `distutils/version.py`, StrictVersion is pretty strict in
what version numbers to accept:

> A version number consists of two or three dot-separated numeric
> components, with an optional "pre-release" tag on the end.  The
> pre-release tag consists of the letter 'a' or 'b' followed by a number.

This assumption already fails for some pretty basic python libraries
itself, like setuptools, also available in `46.1.3.post20200610`, a
completely valid version number according to
https://www.python.org/dev/peps/pep-0440/#post-releases.

There doesn't seem to be a particular reason on why StrictVersion has
been used here, so let's use LooseVersion, to be compatible with these
versions.

## Pull Request Checklist

- [x] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [x] Include your name in `AUTHORS.md` if you like.

flokli commented:
I'm not really sure on how to interpret the travis failure in `modification-check.py`. This is an intended change to `certbot-auto` as well - not sure how to tell travis that's fine. @adferrand?

adferrand commented:
The failure is because output of `letsencrypt-auto-source/build.py` output is not matching the current content of `letsencrypt-auto` script in your PR. Run locally `letsencrypt-auto-source/build.py`, commit and push the new `letsencrypt-auto` generated, and you are good to go.

adferrand commented:
Oh, and do not modify the `certbot-auto` and `letsencrypt-auto` scripts at the root of the repository. They are supposed to be regenerated during a release. Only what is in `letsencrypt-auto-source` is relevant for the development.

flokli commented:
Makes sense, thanks for the explanation! Just updated the PR.

adferrand commented on certbot/certbot/util.py:
```diff
@@ -563,11 +563,11 @@ def get_strict_version(normalized):
     """
     # strict version ending with "a" and a number designates a pre-release
-    return distutils.version.StrictVersion(normalized.replace(".dev", "a"))
+    return distutils.version.LooseVersion(normalized.replace(".dev", "a"))
```
This function is part of Certbot public API and so its behavior should not change.

flokli replied:
Agreed, removed.

flokli commented:
Thanks for the review.

I agree not modifying `{certbot,letsencrypt}-auto` is fine. What's was really causing troubles here were the `setup.py` files.

I updated my PR to only include changes there:
```
 AUTHORS.md                          | 1 +
 acme/setup.py                       | 4 ++--
 certbot-apache/setup.py             | 4 ++--
 certbot-ci/setup.py                 | 4 ++--
 certbot-compatibility-test/setup.py | 4 ++--
 certbot-dns-cloudflare/setup.py     | 4 ++--
 certbot-dns-cloudxns/setup.py       | 4 ++--
 certbot-dns-digitalocean/setup.py   | 4 ++--
 certbot-dns-dnsimple/setup.py       | 4 ++--
 certbot-dns-dnsmadeeasy/setup.py    | 4 ++--
 certbot-dns-gehirn/setup.py         | 4 ++--
 certbot-dns-google/setup.py         | 4 ++--
 certbot-dns-linode/setup.py         | 4 ++--
 certbot-dns-luadns/setup.py         | 4 ++--
 certbot-dns-nsone/setup.py          | 4 ++--
 certbot-dns-ovh/setup.py            | 4 ++--
 certbot-dns-rfc2136/setup.py        | 4 ++--
 certbot-dns-route53/setup.py        | 4 ++--
 certbot-dns-sakuracloud/setup.py    | 4 ++--
 certbot-nginx/setup.py              | 4 ++--
 certbot/CHANGELOG.md                | 3 +++
 certbot/setup.py                    | 4 ++--
 22 files changed, 44 insertions(+), 40 deletions(-)
```

… so I dropped the changes from `certbot/certbot/util.py`, `letsencrypt-auto[.template]` and `pipstrap.py`.

flokli commented:
Thanks a lot!
